main/src/ca/uwaterloo/flix/api/lsp/provider/ShowAstProvider.scala

```diff
@@ -32,17 +33,14 @@ object ShowAstProvider {
     * - `title` (a string like `Namer.flix.ir`)
     * - `text` (a string with the ir representation).
     */
-  def showAst(phase: String)(implicit index: Index, root: Option[Root], flix: Flix): JObject = root match {
-    case None =>
-      val text = "No IR available. Does the program not compile?"
-      astObject(phase, text)
-    case Some(r) =>
-      // We have to compile the program to obtain the relevant AST.
-      flix.codeGen(r)
-
-      val phasesWithNames = AstPrinter.allPhases().map { case (name, printer) => (name, (name, printer)) }
-      val (closestPhase, closestPrettyPrinter) = Similarity.closestMatch(phase, phasesWithNames)
-      astObject(closestPhase, closestPrettyPrinter())
+  def showAst(phase: String)(implicit index: Index, root: Option[Root], flix: Flix): JObject = {
+    val names = AstPrinter.phaseNames().m1.keys.map(s => (s, s)).toMap
+    val closestName = Similarity.closestMatch(phase, names)
+    val oldOpts = flix.options
+    flix.setOptions(oldOpts.copy(xprintphase = Set(closestName)))
+    flix.compile()
+    flix.setOptions(oldOpts)
+    astObject(closestName, s"Check in build/asts/${closestName}.flixir")
```

Comment:
It works, but it breaks `ShowAst`. Its not easy to hold onto the string so it can be given to lsp, so maybe the easier choice is just that the compiler should give vscode the path to the ast file, which vscode can then open? instead of giving the string back directly

Reply:
Interesting approach. How about you refactor towards that.

@sockmaster27 Can we make it so that ShowAst get passed back a path which it then opens?

Reply:
I don't see why not

Reply:
@JonathanStarup Can you make the changes to the compiler? Then once that is in place, @sockmaster27 can make the changes to VSCode.

Reply:
New plan: show ast provider just returns a path where they all live